hw8/hw8_t02.py

```diff
@@ -0,0 +1,64 @@
+"""
+Task.
+
+Write a wrapper class TableData for database table, that when initialized with database name and
+table acts as collection object (implements Collection protocol).
+Assume all data has unique values in 'name' column.
+So, if presidents = TableData(database_name='example.sqlite', table_name='presidents')
+
+then
+ -  `len(presidents)` will give current amount of rows in presidents table in database
+ -  `presidents['Yeltsin']` should return single data row for president with name Yeltsin
+ -  `'Yeltsin' in presidents` should return if president with same name exists in table
+ -  object implements iteration protocol. i.e. you could use it in for loops::
+       for president in presidents:
+           print(president['name'])
+ - all above mentioned calls should reflect most recent data.
+ If data in table changed after you created collection instance, your calls should return updated data.
+
+Avoid reading entire table into memory.
+ When iterating through records, start reading the first record, then go to the next one,
+  until records are exhausted.
+When writing tests, it's not always necessary to mock database calls completely.
+ Use supplied example.sqlite file as database fixture file.
+"""
+
+import sqlite3
+
+
+class TableData:
+    def __init__(self, db_path: str, table_name: str):
+        self.db_path = db_path
+        self.table_name = (table_name,)
+
+    def connect_and_return_cursor(self) -> sqlite3.connect().cursor:
+        connection = sqlite3.connect(self.db_path)
+        return connection.cursor()
+
+    def __len__(self):
+        self.cursor = self.connect_and_return_cursor()
+        self.cursor.execute(f"SELECT count(*) FROM {self.table_name[0]}")  # noqa: S608
+        return self.cursor.fetchone()[0]
+
+    def __iter__(self):
+        self.cursor = self.connect_and_return_cursor()
+        for row in self.cursor.execute(
+            f"SELECT * FROM {self.table_name[0]}"  # noqa: S608
+        ):  # noqa: S608
+            yield {"name": row[0], "age": row[1], "country": row[2]}
```

Comment:
Invented to yield a dict to fit 
`       for president in presidents:
           print(president['name'])`
requirement...

Reply:
I just got familiar with `sqlite3.Row` objects! Need a lil more time